### PR TITLE
Feature/GitHub

### PR DESCRIPTION
**Pull request by arfbtwn**
Took me a few tries to get the formatting right - this is effectively just the HACKING file from the main part of the repository with some GH specific markdown

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+     of the copyright owners of a file if you're modifying a considerable
+     proportion of it.
+
+
+C# Coding Style Guidelines
+==========================
+
+These guidelines should be followed when writing code in Banshee. For the most
+part they are similar to the Mono syntax guidelines [1]. All public API must
+adhere to the .NET Framework Design Guidelines. [2]
```
Comment (Carbenium) [inline .github/CONTRIBUTING.md]: nit: [2] before the dot.
Is it possible to wrap references in [] or () ? In the end it should look like [2].

(Sorry, I have a scientific background :wink:  )

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+Patches and additions to the code base will be checked for adherence to these
+guidelines. If code is in violation, you will be asked to reformat it.
+
+  1. Private variable/field names should be written like:
+
+      lower_case_with_under_scores
```
Comment (Carbenium) [inline .github/CONTRIBUTING.md]: I'd really like to see this one changed. Not sure if it's feasible.

`testVar` just looks more natural to me in C#. Python would be something else...

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+  2. Property, event, and method names should be written like:
+
+      UpperCaseStartingLetter
+
+  3. A space before method/conditional parenthesis, braces:
+
+```C#
+if (condition) {
+    CallSomeFunction (args);
```
Comment (Carbenium) [inline .github/CONTRIBUTING.md]: The space feels strange, too.

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+     equality, mathematical, event-subscription, ...):
+
+```C#
+var compare = (a + b * c) != (d - e * f);
+```
+
+  12. Please write `private` accessors even if the language defaults
+    to it in the context you're in.
+
+  13. For tests, use the Assert.That(actual, Is.EqualTo(expected)); syntax,
```
Comment (Carbenium) [inline .github/CONTRIBUTING.md]: nit: code block

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+      Banshee.Cdrom.Gui
+      Banshee.Cdrom.Nautilus
+
+  2. Try to keep GUI as separate as possible from "real work" and keep
+     the namespace separate as well, if possible and applicable. For instance,
+     Many different CD-ROM backends could be written for different 
+     platforms, but the same GUI should be used. Don't put GUI code in
+     the platform implementation:
+
+      Banshee.Cdrom
```
Comment (Carbenium) [inline .github/CONTRIBUTING.md]: nit: code block

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+  6. Avoid calls to Assembly.GetTypes as memory resulting from these calls
+     will not be GCed.
+
+
+Organization Guidelines
+=======================
+
+  1. Organize code into logical namespaces:
+
+      Banshee.Cdrom
```
Comment (Carbenium) [inline .github/CONTRIBUTING.md]: nit: code block

File: .github/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+     generic API must be designed first and then the platform implementation
+     of the API can be added. See Banshee.Configuration for ideas.
+   
+  4. Do not hard code path separators. Use Path.DirectorySeparatorChar instead
+     as it is portable to other platforms. 
+
+  5. Try not to perform many string concatenations. Use a StringBuilder if
+     necessary
+   
+  6. Avoid calls to Assembly.GetTypes as memory resulting from these calls
```
Comment (Carbenium) [inline .github/CONTRIBUTING.md]: nit: code block

File: .github/CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+     Gnome.Vfs IO into top-level APIs.
+
+  3. When a platform-specific task needs to be performed, a top-level, 
+     generic API must be designed first and then the platform implementation
+     of the API can be added. See Banshee.Configuration for ideas.
+   
+  4. Do not hard code path separators. Use Path.DirectorySeparatorChar instead
+     as it is portable to other platforms. 
+
+  5. Try not to perform many string concatenations. Use a StringBuilder if
```
Comment (Carbenium) [inline .github/CONTRIBUTING.md]: nit: code block

File: .github/CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+
+  2. In *most* cases Banshee.IO should be used (and possibly extended) when
+     IO must be performed. Do *NOT* hard-code System.IO, Mono.Unix, or
+     Gnome.Vfs IO into top-level APIs.
+
+  3. When a platform-specific task needs to be performed, a top-level, 
+     generic API must be designed first and then the platform implementation
+     of the API can be added. See Banshee.Configuration for ideas.
+   
+  4. Do not hard code path separators. Use Path.DirectorySeparatorChar instead
```
Comment (Carbenium) [inline .github/CONTRIBUTING.md]: nit: code block

File: .github/CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+
+      Use List<T> instead of ArrayList, Dictionary<K, V> instead of Hashtable
+
+  2. In *most* cases Banshee.IO should be used (and possibly extended) when
+     IO must be performed. Do *NOT* hard-code System.IO, Mono.Unix, or
+     Gnome.Vfs IO into top-level APIs.
+
+  3. When a platform-specific task needs to be performed, a top-level, 
+     generic API must be designed first and then the platform implementation
+     of the API can be added. See Banshee.Configuration for ideas.
```
Comment (Carbenium) [inline .github/CONTRIBUTING.md]: nit: code block

File: .github/CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+=========================
+
+  1. Use generics and generic collections when possible in place of 
+     1.0 features. New code in Banshee should leverage 2.0 features 
+     as much as possible, and old code should be updated as development
+     occurs in a given area.
+
+      Use List<T> instead of ArrayList, Dictionary<K, V> instead of Hashtable
+
+  2. In *most* cases Banshee.IO should be used (and possibly extended) when
```
Comment (Carbenium) [inline .github/CONTRIBUTING.md]: nit: code block

File: .github/CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+
+Implementation Guidelines
+=========================
+
+  1. Use generics and generic collections when possible in place of 
+     1.0 features. New code in Banshee should leverage 2.0 features 
+     as much as possible, and old code should be updated as development
+     occurs in a given area.
+
+      Use List<T> instead of ArrayList, Dictionary<K, V> instead of Hashtable
```
Comment (Carbenium) [inline .github/CONTRIBUTING.md]: nit: code block

File: .github/CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,222 @@
...
+  4. If an acronym is three or more characters long, only its first letter
+     should be capitalized
+
+      Banshee.Cdrom
+      Banshee.Playlists.Formats.Pls
+      Banshee.Playlists.Formats.M3u
+
+  5. Prefix interfaces with 'I'
+
+      IPlaylist
```
Comment (Carbenium) [inline .github/CONTRIBUTING.md]: nit: code block

File: .github/CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,222 @@
...
+     accepted name that is generally well known
+
+  3. If an acronym is one-two characters long, it should be all caps 
+
+      Banshee.IO and not Banshee.Io
+
+  4. If an acronym is three or more characters long, only its first letter
+     should be capitalized
+
+      Banshee.Cdrom
```
Comment (Carbenium) [inline .github/CONTRIBUTING.md]: nit: code block

File: .github/CONTRIBUTING.md [hunk 15]
```diff
@@ -0,0 +1,222 @@
...
+
+  1. Member names should be descriptive and it is best to avoid abbreviations
+     and acronyms
+
+  2. If an abbreviation or acronym is used, it should be in the form of an 
+     accepted name that is generally well known
+
+  3. If an acronym is one-two characters long, it should be all caps 
+
+      Banshee.IO and not Banshee.Io
```
Comment (Carbenium) [inline .github/CONTRIBUTING.md]: nit: code block

File: .github/CONTRIBUTING.md [hunk 16]
```diff
@@ -0,0 +1,222 @@
...
+    to it in the context you're in.
+
+  13. For tests, use the Assert.That(actual, Is.EqualTo(expected)); syntax,
+      which is more readable, and prevents you from misplacing the expected
+      parameter in the place of the actual parameter (a mistake that is very
+      usual when using the older Assert.AreEqual(,) syntax).
+
+  14. Parameter names should be written like:
+
+      camelCaseWord
```
Comment (Carbenium) [inline .github/CONTRIBUTING.md]: nit: code block

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (arfbtwn) [inline .github/CONTRIBUTING.md]: I'm sure you meant the line above and that it should come after the full stop. (oh dear, you're not an emacs user too are you? :trollface:)

On the brackets, yeah, I had intended them to be in there :+1:

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (arfbtwn) [inline .github/CONTRIBUTING.md]: haha yeah, and I'd also like to see the `_` prefix for member variables

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (arfbtwn) [inline .github/CONTRIBUTING.md]: This one you might come to like after a while and once you've changed your IDE settings so you're not fighting it...

File: .github/CONTRIBUTING.md (hunk 15 above)
Comment (arfbtwn) [inline .github/CONTRIBUTING.md]: yeah I wasn't sure with these ones :+1:

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (Carbenium) [inline .github/CONTRIBUTING.md]: Nope. See http://libguides.williams.edu/citing/acs (Section How to Format In-Text Citations)
References in-line references should be in front of the full stop.
But I'm fine with the other style, too. After all it's not a chemistry paper :smile: 

vim it is. :1st_place_medal:

**Comment by arfbtwn** [comment]
this is looking pretty good now but I want to add a small paragraph on reporting issues to bugzilla as well

**Comment by Carbenium** [comment]
Formatting looks good :+1: I also agree on the Bugzilla part. After all it is still the official Banshee bug tracker.

File: .github/CONTRIBUTING.md [hunk 17]
```diff
@@ -0,0 +1,231 @@
+This document defines many guidelines that should be adhered to when developing 
+against Banshee. These guidelines will make the codebase more readable,
+extensible, and portable.
+
+Reporting Issues
+================
+
+[GNOME's bugzilla](https://bugzilla.gnome.org/buglist.cgi?quicksearch=product%3Abanshee)
+is the official issue tracker and contributors are encouraged to file bugs and
+feature requests there in addition to this repository.
```
Comment (arfbtwn) [inline .github/CONTRIBUTING.md]: The wording in this document was already pretty heavy handed (_adhered to_ above and _violation_ of coding guidelines below) so I wanted to soften this a bit - hence _encouraged_ here.

I wanted to highlight that we're trying to provide an adapter on Banshee's cathedral development model with our repositories but I'm not sure this is the best place to do it - the file's already quite long - so I'm considering extending this pull request with a `README.md` where we can say that and link to the original Banshee `README` for further information.

What do you think?

Comment (Carbenium) [inline .github/CONTRIBUTING.md]: IMHO the length is still okay.
If you want to split it, you could consider putting the parts into the docs folder and then link to them.

**Pull request merged**